### PR TITLE
check and link against libatomic

### LINKED ISSUE #171: Linking error (test_atomics_executable) on Raspbian/armv6l
I am trying to compile ROS dashing from source on Raspbian Buster (`gcc (Raspbian 8.3.0-6+rpi1) 8.3.0`) running on armv6l and encounter a linking error for target `test_atomics_executable`:
```
/usr/bin/ld: CMakeFiles/test_atomics_executable.dir/test/test_atomics.c.o: in function `main':
test_atomics.c:(.text+0x11a8): undefined reference to `__atomic_store_8'
/usr/bin/ld: test_atomics.c:(.text+0x11c0): undefined reference to `__atomic_load_8'
/usr/bin/ld: test_atomics.c:(.text+0x1268): undefined reference to `__atomic_store_8'
/usr/bin/ld: test_atomics.c:(.text+0x12b8): undefined reference to `__atomic_exchange_8'
[...]
```

I am using the standard repo at https://raw.githubusercontent.com/ros2/ros2/release-latest/ros2.repos, excluding `rviz2` and `rqt`.

Edit: The problem can be resolved by manually adding `-latomic`, i.e. `target_link_libraries(test_atomics_executable ${PROJECT_NAME} -latomic)`.

### PR DESCRIPTION
**Pull request by christianrauch**
Link against libatomic. Fixes #171 .

### CODE REVIEW

**Comment by christianrauch** [comment]
@clalancette @sloretz Can you have a look at this?
I would like to propose the same linking fix for a couple of other packages where I encountered the same `libatomic` linking issue.

File: CMakeLists.txt [hunk 1]
```diff
@@ -214,6 +218,10 @@ if(BUILD_TESTING)
       LANGUAGE C
   )
   target_link_libraries(test_atomics_executable ${PROJECT_NAME})
+  if(HAVE_LIBATOMICS)
+    target_link_libraries(test_atomics_executable -latomic)
```
[the remark this answers is not in the document]
Comment (christianrauch) [reply CMakeLists.txt]: Thanks, I didn't know that.

**Comment by christianrauch** [comment]
I got an error with `ament_export_library_names(atomic)`, but `ament_export_libraries(atomic)` did work.

**Comment by sloretz** [comment]
CI (building all, testing just rcutils)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8040/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4048/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6548/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7900/)

**Comment by sloretz** [comment]
Thanks for the PR @christianrauch !

**Comment by dirk-thomas** [comment]
Every package using `rcutils` reports the following CMake warning for me locally:

```
CMake Warning at <ws>/install/rcutils/share/rcutils/cmake/ament_cmake_export_libraries-extras.cmake:116 (message):
  Package 'rcutils' exports library 'atomic' which couldn't be found
Call Stack (most recent call first):
  <ws>/install/rcutils/share/rcutils/cmake/rcutilsConfig.cmake:38 (include)
  CMakeLists.txt:34 (find_package)
```

**Comment by sloretz** [comment]
> Every package using rcutils reports the following CMake warning for me locally:

Looks like I forgot to include a repos URL in CI. Re-running to check if it affects the build farm too

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8042/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4049/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6549/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7905/)

**Pull request merged**